src/internal/collection/postgres_collection.go

```diff
@@ -351,30 +430,72 @@ func (c *postgresCollection) list(
 		return nil
 	}
 
-	result := &model{}
-	for rows.Next() {
-		if err := rows.StructScan(result); err != nil {
-			return c.mapSQLError(err, "")
+	// To avoid holding the DB connection open for an unknown duration
```

Comment:
This is a bit pedantic, but we should probably be more precise about the fact that we are opening / closing a transaction during the read rather than opening / closing a database connection. The database connection to pgbouncer will remain open, but by finishing the transaction we will ensure that a real database connection is freed up for a different client.